hammer/par/innovus/__init__.py

```diff
@@ -1065,14 +1066,19 @@ def generate_floorplan_tcl(self) -> List[str]:
                         current_top_layer = None
                     if current_top_layer is not None:
                         bot_layer = self.get_stackup().get_metal_by_index(1).name
-                        cover_layers = list(map(lambda m: m.name, self.get_stackup().get_metals_below_layer(current_top_layer)))
-                        output.append("create_place_halo -insts {inst} -halo_deltas {{{s} {s} {s} {s}}} -snap_to_site".format(
-                            inst=new_path, s=spacing))
-                        output.append("create_route_halo -bottom_layer {b} -space {s} -top_layer {t} -inst {inst}".format(
+                        cover_layers = list(map(lambda m: m.name, self.get_stackup().get_metals_incl_layer(current_top_layer)))
```

Comment:
Doesn't this prevent power straps by abutment on the top layer?

Reply:
It would, but I'm not sure how we could avoid the current issue if pins are also generated on the top layer as well.

Reply:
Oh, I missed that. Fundamentally, pins should not be generated on the power top layer. I would advocate for enforcing this in `place_pins` instead of doing this halo-ing.

Reply:
Sounds good, so I will add code to:
1) Stop signal pins from being generated on the top layer of a macro. 
2) Implement the haloing scheme only for layers below the top layer.

Reply:
I think it should be a logger error but not kill the run (in case it is intentional).

Reply:
Added logger error when power pins are found on same layer as signal pins.